noether_examples/src/halfedge_finder_node.cpp

```diff
@@ -0,0 +1,198 @@
+/**
+ * @author Jorge Nicho <jrgnichodevel@gmail.com>
+ * @file halfedge_finder_node.cpp
+ * @date Dec 5, 2019
+ * @copyright Copyright (c) 2019, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <ros/ros.h>
+#include <boost/format.hpp>
+#include <boost/filesystem.hpp>
+#include <noether_conversions/noether_conversions.h>
+#include <visualization_msgs/MarkerArray.h>
+#include <noether_msgs/GenerateEdgePathsAction.h>
+#include <actionlib/client/simple_action_client.h>
+#include <console_bridge/console.h>
+
+using RGBA = std::tuple<double,double,double,double>;
```

Comment:
Should you std::array here since they are the same type. They it would have the same benefits as a vector just fixed size like iterators and the operator[]?

Reply:
I agree but changing this would involve changing various other repos that use the function that takes a tuple.